### ISSUE TITLE
Schedule: SelectEvent does not bring the correct time

### ISSUE THREAD

**Issue by nokiajavi** [primefaces/primefaces#11898]
### Describe the bug

I am in the Europe/Madrid zone, and I have the schedule configured with that zone, but when I click to create a new event, for example at 9:00 AM, the event brings the wrong time, in this case 5:00 A.M.

I have tried configuring different ways, and the only one that gives me the correct time is if I set GMT-2  in the attribute **timeZone**. Currently in Spain, we are in GMT+2, the problem with this solution is that when the time zone changes to GMT+1 this fix will no longer work for me.
![schedule](https://github.com/primefaces/primefaces/assets/13865234/d78d5b38-7454-40f8-aaa2-94c778f0fdb4)

### Reproducer

1 Select the day view of the component
2. Then click on some time
3. In the ajax event method, show a dialog with the event data.

HTML Code:
`<p:schedule id="idAgenda" value="#{form.agendaModel}" locale="#{text['calendar.language']}" slotDuration="01:00:00" 
	        		    		draggable="false" resizable="false" slotLabelFormat="HH:mm" timeFormat="HH:mm"
	        		    		allDaySlot="false" minTime="09:00:00" maxTime="21:00:00" showWeekends="false" scrollTime="09:00:00" clientTimeZone="local" timeZone="#{form.timerZone}">
		        		    	<p:ajax event="viewChange" listener="#{form.cambiaVistaAgenda}" update="idAgenda" />
								<p:ajax event="eventSelect" listener="#{form.seleccionaEvento}" resetValues="true"/>
								<p:ajax event="dateSelect" listener="#{form.seleccionaFecha}" resetValues="true" />
		        		    </p:schedule>`

Content Dialog code:
```
<p:outputLabel id="labeldia" value="#{text['label.dia']}: " />
		        		<p:calendar id="idFechaCita" for="labeldia" value="#{form.cita.fechcita}" showButtonPanel="true" locale="#{text['calendar.language']}"  required="true"
							pattern="#{text['date.format']}" readonlyInput="false" placeholder="dd/mm/aaaa" mask="99/99/9999" maskAutoClear="false" maskSlotChar="dd/mm/aaaa" 
							label="#{text['label.dia']}" maxdate="c+365" inputStyleClass="align_centro margenSuperior-5" selectOtherMonths="true"  showOtherMonths="true" navigator="true">
								<p:ajax partialSubmit="true" process="@none" event="viewChange" onstart="guardarPosicionCalendario()" oncomplete="restaurarPosicionCalendario()"/>
						</p:calendar>
		           <p:outputLabel id="labelhora" value="#{text['label.hora']}: " />
		            <p:calendar id="idHora" widgetVar="idHoraWidget" for="labelhora" value="#{form.cita.horacita}" showButtonPanel="true" locale="#{text['calendar.language']}" required="true" 
							 pattern="HH:mm" mask="true" maskSlotChar="HH:mm" timeOnly="true" label="#{text['label.hora']}" inputStyleClass="align_centro margenSuperior-5">
					</p:calendar>
```

Java code Form:
`public TimeZone getTimerZone() {
		return TimeZone.getTimeZone("Europe/Madrid");
	}`

Ajax seleccionaFecha:
```
public void seleccionaFecha(SelectEvent evento){
    	PrimeFaces pfidialog = PrimeFaces.current();
    	//Capturamos el elemento agenda del xhtml para poder trabajar con el
	Schedule miAgenda = (Schedule) FacesContext.getCurrentInstance().getViewRoot().findComponent(FORM_ID_AGENDA);
	if(miAgenda != null && evento != null ) {
		   DefaultScheduleEvent eventoSeleccionado = new DefaultScheduleEvent("", LocalDateTime.fromDateFields((Date) evento.getObject()).toDate(),LocalDateTime.fromDateFields((Date) evento.getObject()).toDate());
		    		eventoDialog= new EventoDialog();
		    		eventoDialog.setFechaHora(LocalDateTime.fromDateFields((Date) evento.getObject()).toDate());
		    	}
		    	pfidialog.ajax().update("citaDialogId");
		    	pfidialog.executeScript("PF('citaDialog').show();");	
} 
```

### Expected behavior

The SelectEvent event should bring the correct time, based on the set timerZone.

### PrimeFaces edition

Community

### PrimeFaces version

7.0

### JSF implementation

Mojarra

### JSF version

2.3.2

### Java version

1.8

**Comment by melloware** [comment]
@nokiajavi is everything else working properly and its just the SelectEvent that has the wrong time?  Just trying to get a clear picture of the bug.

**Comment by melloware** [comment]
@nokiajavi i am pretty sure this was fixed in 12.0.0+ see: https://github.com/primefaces/primefaces/issues/8076

**Comment by nokiajavi** [comment]
> @nokiajavi is everything else working properly and its just the SelectEvent that has the wrong time? Just trying to get a clear picture of the bug.
> 
> Also you are on 7.0 have you tried 14.0.0. There is a good chance this bug is fixed.
> 
> Also you can see on our showcase we let you set the `timezone` and `clientTimeZone` to test it.
> 
> https://www.primefaces.org/showcase/ui/data/schedule/configuration.xhtml?jfwid=7f446

Sorry for my mistake @melloware , I was referring to the "DateSelect" event, not "SelectEvent", the "EventSelect" works correctly.

Currently it is impossible for me to migrate even to PF8, I have tested it in the showcase and it works fine, it is possible that, as you indicate, it has been corrected subsequently, since in the PF8 showcase it works correctly.

I have been reviewing the ticket that you mentioned, I have downloaded the example project and I have put PF7.0 in it and I have tested the component by putting in the variables:
clientTimeZone="Europe/Madrid" timeZone="Europe/Madrid"

It works, so I'll review my code, because I remember that if I didn't put "local" in clientTimeZone, all the events that I already had loaded from DB would change their time. Possibly they have been saved with the wrong time in the database.

Thank you very much for everything as always!
